symbolic-symcache/src/compat.rs

```diff
@@ -0,0 +1,288 @@
+        }
+    }
+    /// Returns whether this cache is up-to-date.
+    pub fn is_latest(&self) -> bool {
+        self.version() == new::raw::SYMCACHE_VERSION
+    }
+
+    /// The architecture of the symbol file.
+    pub fn arch(&self) -> Arch {
+        match &self.0 {
+            SymCacheInner::New(symc) => symc.arch(),
+            SymCacheInner::Old(symc) => symc.arch(),
+        }
+    }
+
+    /// The debug identifier of the cache file.
+    pub fn debug_id(&self) -> DebugId {
+        match &self.0 {
+            SymCacheInner::New(symc) => symc.debug_id(),
+            SymCacheInner::Old(symc) => symc.debug_id(),
+        }
+    }
+
+    /// Returns true if line information is included.
+    pub fn has_line_info(&self) -> bool {
+        match &self.0 {
+            SymCacheInner::New(symc) => symc.has_line_info(),
+            SymCacheInner::Old(symc) => symc.has_line_info(),
+        }
+    }
+
+    /// Returns true if file information is included.
+    pub fn has_file_info(&self) -> bool {
+        match &self.0 {
+            SymCacheInner::New(symc) => symc.has_file_info(),
+            SymCacheInner::Old(symc) => symc.has_file_info(),
+        }
+    }
+
+    /// Returns an iterator over all functions.
+    pub fn functions(&self) -> Functions<'data> {
+        match &self.0 {
+            SymCacheInner::New(symc) => Functions(FunctionsInner::New(symc.functions())),
+            SymCacheInner::Old(symc) => Functions(FunctionsInner::Old(symc.functions())),
+        }
+    }
+
+    /// Given an address this looks up the symbol at that point.
+    ///
+    /// Because of inline information this returns a vector of zero or
+    /// more symbols.  If nothing is found then the return value will be
+    /// an empty vector.
+    pub fn lookup(&self, addr: u64) -> Result<Lookup<'data, '_>, SymCacheError> {
+        match &self.0 {
+            SymCacheInner::New(symc) => Ok(Lookup(LookupInner::New {
+                iter: symc.lookup(addr),
+                lookup_addr: addr,
+            })),
+            SymCacheInner::Old(symc) => {
+                let lookup = symc.lookup(addr)?;
+                Ok(Lookup(LookupInner::Old(lookup)))
+            }
+        }
+    }
+}
+
+impl<'slf, 'd: 'slf> AsSelf<'slf> for SymCache<'d> {
+    type Ref = SymCache<'slf>;
+
+    fn as_self(&'slf self) -> &Self::Ref {
+        self
+    }
+}
+
+#[derive(Clone, Debug)]
+enum FunctionInner<'data> {
+    Old(old::Function<'data>),
+    New(new::Function<'data>),
+}
+
+/// A function in a `SymCache`.
+#[derive(Clone, Debug)]
+pub struct Function<'data>(FunctionInner<'data>);
+
+impl<'data> Function<'data> {
+    /// The ID of the function.
+    pub fn id(&self) -> usize {
+        match &self.0 {
+            FunctionInner::Old(function) => function.id(),
+            // TODO: Is there something better we can return here?
+            // I doubt anyone actually cares about this.
+            FunctionInner::New(_) => usize::MAX,
```

Comment:
I agree, probably noone will care. But I still think we should at least return something unique, which in that case would just be the function index.